modules/healthcheck/default.nix

```diff
@@ -0,0 +1,281 @@
+{ withSystem, ... }:
+{
+  flake.modules.nixos.healthcheck =
+    # /etc/nixos/modules/systemd-healthcheck.nix
+    {
+      config,
+      options,
+      lib,
+      pkgs,
+      ...
+    }:
+
+    with lib;
+
+    let
+      # Options for the liveness probe (timer-based check)
+      mkLivenessProbeOptions = {
+        options = {
+          enable = mkEnableOption "the liveness probe";
+
+          command = mkOption {
+            type = types.lines;
+            description = "The command to execute for the liveness check.";
+          };
+
+          initialDelaySeconds = mkOption {
+            type = types.int;
+            default = 15;
+            description = "Seconds to wait after the service is up before the first liveness probe.";
+          };
+
+          periodSeconds = mkOption {
+            type = types.int;
+            default = 30;
+            description = "How often (in seconds) to perform the probe.";
+          };
+
+          timeoutSeconds = mkOption {
+            type = types.int;
+            default = 10;
+            description = "Seconds after which the liveness probe command times out.";
+          };
+        };
+      };
+
+      # Options for the readiness probe (notify-based check)
+      mkReadinessProbeOptions = {
+        options = {
+          enable = mkEnableOption "the notify readiness probe";
+
+          command = mkOption {
+            type = types.lines;
+            description = "The check command to run in a loop. It should output a value on success.";
+            example = "curl -s localhost:9091/metrics | awk '/^signing_signers_loaded_count /{print int($2)}'";
+          };
+
+          periodSeconds = mkOption {
```

Comment:
Maybe `intervalSeconds`? Same for the same-named option for the liveness probe up above. Maybe also consider dropping the `seconds` suffix and specifying the time unit in the description of all the relevant options.